### PR TITLE
Pdf generator

### PR DESCRIPTION
**Pull request by garethgooch**

### CODE REVIEW

File: test/models/SubmissionSpec.scala [hunk 1]
```diff
@@ -18,42 +18,102 @@ package models
...
   }
 
   ".asMap" must {
 
     "return a map" in {
-      when(answers.selectTaxYear) thenReturn Some(SelectTaxYear.CYMinus2)
-      val submission = Submission(answers)
+      val TimeNow = DateTime.now()
+      DateTimeUtils.setCurrentMillisFixed(TimeNow.getMillis)
+      val timedAnswers = MockUserAnswers.minimalValidUserAnswers
```
Comment (Pascoe121) [inline test/models/SubmissionSpec.scala]: same as point above.

Comment (garethgooch) [reply test/models/SubmissionSpec.scala]: Missed in commit

File: test/models/SubmissionSpec.scala [hunk 2]
```diff
@@ -18,42 +18,102 @@ package models
...
 
   ".apply" must {
 
     "build " in {
-      when(answers.selectTaxYear) thenReturn Some(SelectTaxYear.CYMinus2)
-      val submission = Submission(answers)
+      val TimeNow = DateTime.now()
+      DateTimeUtils.setCurrentMillisFixed(TimeNow.getMillis)
+      val timedAnswers = MockUserAnswers.minimalValidUserAnswers
+      when(timedAnswers.selectTaxYear) thenReturn Some(SelectTaxYear.CYMinus2)
```
[the remark this answers is not in the document]
Comment (garethgooch) [reply test/models/SubmissionSpec.scala]: Missed in commit

File: app/models/Submission.scala [hunk 3]
```diff
@@ -29,14 +30,27 @@ object Submission {
...
+      taxYear =>
+        taxYear.asString
+    ).getOrElse("Failed to get Tax year")
+
+    Submission(taxYear, pdf, meta)
 
-    Submission(answers.selectTaxYear.get)
   }
 
   def asMap(e: Submission): Map[String, String] = {
```
[the remark this answers is not in the document]
Comment (garethgooch) [reply app/models/Submission.scala]: Missed in commit

File: app/models/templates/Metadata.scala [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+  implicit def reads: Reads[Metadata] = (
+      (__ \ "customerId").read[String] and
+      (__ \ "hmrcReceivedAt").read[LocalDateTime](jodaDateReads) and
+      (__ \ "xmlCreatedAt").read[LocalDateTime](jodaDateReads)
+    )(Metadata.apply(_,_,_))
+
+
+  def apply(customerId: String, hmrc: String): Metadata = {
```
[the remark this answers is not in the document]
Comment (garethgooch) [reply app/models/templates/Metadata.scala]: Amended

File: app/controllers/CheckYourAnswersController.scala [hunk 5]
```diff
@@ -47,7 +51,16 @@ class CheckYourAnswersController @Inject()(appConfig: FrontendAppConfig,
...
       implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
 
-      submissionService.ctrSubmission(request.userAnswers) map {
+      val cyaHelper = new CheckYourAnswersHelper(request.userAnswers)
+      val sections = new CheckYourAnswersSections(cyaHelper, request.userAnswers)
+      val pdfHtml = pdf_check_your_answers(appConfig, sections.sectionsToShow)
+      dataCacheConnector.save[String](request.externalId, "pdfHtml", pdfHtml.toString())
+
+      implicit val metadata = new Metadata("")
+      dataCacheConnector.save(request.externalId, "metadata", metadata)
```
[the remark this answers is not in the document]
Comment (garethgooch) [reply app/controllers/CheckYourAnswersController.scala]: The model contains the metadata, this is just adding this at submission so we have the time stamps, have removed the quotes as not required.

**Pull request merged**